roachtest: sqlsmith/setup=tpch-sf1/setting=no-mutations failed

roachtest.sqlsmith/setup=tpch-sf1/setting=no-mutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3311258&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3311258&tab=artifacts#/sqlsmith/setup=tpch-sf1/setting=no-mutations) on release-21.1 @ [147ff1527172ca8ce2d190594d098314af1c2ab6](https://github.com/cockroachdb/cockroach/commits/147ff1527172ca8ce2d190594d098314af1c2ab6):


```
The test failed on branch=release-21.1, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=tpch-sf1/setting=no-mutations/run_1
	sqlsmith.go:205,sqlsmith.go:236,test_runner.go:733: error: pq: internal error: unexpected error from the vectorized engine: runtime error: slice bounds out of range [:1022] with capacity 1021
		stmt:
		SELECT
			count(*) AS col_627
		FROM
			defaultdb.public.lineitem@[0] AS tab_291
		GROUP BY
			tab_291.l_orderkey, tab_291.l_linenumber
		ORDER BY
			tab_291.l_orderkey, tab_291.l_linenumber DESC
		LIMIT
			93:::INT8;
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh sqlsmith/setup=tpch-sf1/setting=no-mutations
```
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=tpch-sf1/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/colexec/ordered_aggregator.go
+++ b/pkg/sql/colexec/ordered_aggregator.go
@@ -288,7 +288,8 @@ func (a *orderedAggregator) Next() coldata.Batch {
 				// capacity, so we choose to instantiate the batch with fixed
 				// maximal capacity that can be needed by the aggregator.
 				a.allocator.ReleaseMemory(colmem.GetBatchMemSize(a.scratch.Batch))
-				a.scratch.Batch = a.allocator.NewMemBatchWithFixedCapacity(a.outputTypes, 2*coldata.BatchSize())
+				newMinCapacity = 2 * coldata.BatchSize()
+				a.scratch.Batch = a.allocator.NewMemBatchWithFixedCapacity(a.outputTypes, newMinCapacity)
 			} else {
 				a.scratch.Batch, _ = a.allocator.ResetMaybeReallocate(
 					a.outputTypes, a.scratch.Batch, newMinCapacity, maxBatchMemSize,

